manifests/init.pp

```diff
@@ -231,13 +232,24 @@
     $dburl_real = $dburl
   }
   else {
-    $dburl_real = $db ? {
-      'postgresql' => "jdbc:${db}://${dbserver}:${dbport_real}/${dbname}",
-      'mysql'      => "jdbc:${db}://${dbserver}:${dbport_real}/${dbname}?useUnicode=true&amp;characterEncoding=UTF8&amp;sessionVariables=default_storage_engine=InnoDB",
-      'oracle'     => "jdbc:${db}:thin:@${dbserver}:${dbport_real}:${dbname}",
-      'sqlserver'  => "jdbc:jtds:${db}://${dbserver}:${dbport_real}/${dbname}",
-      'h2'         => "jdbc:h2:file:/${jira::homedir}/database/${dbname}",
+    if oracle_sid {
```

Comment:
How are these tests passing though?  Isn't this missing a leading `$`??

Also, is there a way of implementing this with less repetition?  Maybe by doing something like...
```puppet
$oracle_jdbc_separator = bool2str($oracle_sid, ':', '/')
```

Reply:
Hmm yeah that does seem weird. I changed it to use the function as you suggested.